### PR TITLE
Feat / Featured shelf cosmetic update

### PR DESCRIPTION
**Pull request by royschut**
## Feat / Featured shelf cosmetic update

This PR improves the look and feel of the featured shelf. Instead of using the TileSlider with just a larger Card, this new version leverages the reserved space much more, with a smooth animation between slides. On mobile devices, you can drag the text off screen to bring in the next. 

See the screenshots/screen caps of desktop and mobile below.

### Steps completed:

According to our definition of done, I have completed the following steps:

- [x] Acceptance criteria met
- [x] Unit tests added
- [ ] Docs updated (including config and env variables)
- [x] Translations added
- [x] UX tested
- [x] Browsers / platforms tested
- [x] Rebased & ready to merge without conflicts
- [x] Reviewed own code

Screenshot 2024-10-25 at 10 24 44
Screenshot 2024-10-25 at 10 24 21

https://github.com/user-attachments/assets/b8ac0c1c-0cd4-4c5c-99cc-79a8f66a7554

https://github.com/user-attachments/assets/ef917b90-3b55-4af5-b580-cb5ce3aeb1a2

### CODE REVIEW

**Comment by langemike** [comment]
Really nice improvement! I detected one visual problem; when you use the pagination dots, they move. So i need to reposition my mouse cursor anytime I want to click the dots.

Other points to consider:
1. Curious: why have you chosen to do inline JS/CSS animations as opposed to regular CSS? I think the later makes it easier to make styling changes. I understand it to calculate a dot size, but maybe the rest could be done in regular CSS.
2. I think we should use `aria-disabled` instead of `disabled` on the slider buttons/paginations. So the buttons can still keep their focus, while the screen-reader pronounces it's state. I know this is not the first time we have discussions about this ;-)
3. I think we should wrap the pagination within` <nav>` and `<ul>`. So the screen-reader pronounces the amount of items and makes it easier to navigate.

File: platforms/web/public/locales/en/common.json [hunk 1]
```diff
@@ -27,5 +28,6 @@
   "slide_next": "Next slide",
   "slide_previous": "Previous slide",
   "slide_right": "Slide right",
+  "slide_to": "Slide to page {{page}} of {{pages}}",
```
Comment (langemike) [inline platforms/web/public/locales/en/common.json]: Minor; I would go with "Slide to item x of y" or "Go to item x of y". When we wrap it in an `<ul>`, the "x of y" part is not needed in my eyes.

I think we should only use "pages" if there are more items per slide.

Using "pages" as a metaphor, could give the assumption a page can consist of multiple items. Such as

Comment (royschut) [reply platforms/web/public/locales/en/common.json]: I agree!

**Review by langemike — APPROVED** [review]
Nice work! I'm satisfied with the applied height. I think the pagination alignment can be improved, but this is something for in the future to iterate on. Good enough for now!

File: packages/ui-react/src/components/FeaturedShelf/FeaturedMetadata.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  playlistId: string | undefined;
+  style?: CSSProperties;
+  hidden?: boolean;
+}) => {
+  const navigate = useNavigate();
+  const { t } = useTranslation('common');
+
+  if (!item) return null;
+
+  const hasVideo = item.sources.find((source) => source.file.indexOf('.m3u8') > -1 || source.file.indexOf('.mp4') > -1);
```
Comment (ChristiaanScheermeijer) [inline packages/ui-react/src/components/FeaturedShelf/FeaturedMetadata.tsx:38]: We might need a util for this and also check for the content type. Some non-playable content types may contain dummy streams in the source list.

File: packages/ui-react/src/hooks/useScrolledDown.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import { useEffect, useRef, useState } from 'react';
+import useEventCallback from '@jwp/ott-hooks-react/src/useEventCallback';
+
+export const useScrolledDown = (scrollUpHeight = 300, scrollDownHeight = 30) => {
```
Comment (ChristiaanScheermeijer) [inline packages/ui-react/src/hooks/useScrolledDown.ts:4]: Let's refactor this (later) to a `useScrollProgress(start: number, end: number)` so we can fade the background smooth with the progress. IMHO, this mechanism works only for TV apps (spatial navigation).

File: packages/ui-react/src/components/FeaturedShelf/FeaturedShelf.module.scss [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+.shelf {
+  display: flex;
+  flex-direction: column;
+  justify-content: flex-end;
+  align-items: flex-start;
+  width: 100%;
+  height: 70vh;
+  min-height: 350px;
+  margin: 0 0 24px 0;
+  padding: 100px 56px;
```
Comment (ChristiaanScheermeijer) [inline packages/ui-react/src/components/FeaturedShelf/FeaturedShelf.module.scss]:
```suggestion
  padding: 100px calc(variables.$base-spacing * 3);
```

File: packages/ui-react/src/components/FeaturedShelf/FeaturedShelf.module.scss [hunk 5]
```diff
@@ -0,0 +1,308 @@
...
+  position: fixed;
+  top: 0;
+  right: 0;
+  left: 0;
+  z-index: 0;
+  display: flex;
+  justify-content: flex-end;
+  align-items: flex-end;
+  width: 100%;
+  height: calc(70vh + 56px);
```
Comment (ChristiaanScheermeijer) [inline packages/ui-react/src/components/FeaturedShelf/FeaturedShelf.module.scss]: Is this for the header?

```suggestion
  height: calc(70vh + variables.$header-height);
```

File: packages/ui-react/src/components/FeaturedShelf/FeaturedShelf.module.scss [hunk 6]
```diff
@@ -0,0 +1,308 @@
...
+  background-color: var(--featured-shelf-background-color);
+
+  @include responsive.mobile-and-small-tablet() {
+    height: calc(67vh + 56px);
```
Comment (ChristiaanScheermeijer) [inline packages/ui-react/src/components/FeaturedShelf/FeaturedShelf.module.scss]:
```suggestion
    height: calc(67vh + variables.$header-height);
```

File: packages/ui-react/src/components/FeaturedShelf/FeaturedShelf.module.scss [hunk 7]
```diff
@@ -0,0 +1,308 @@
...
+}
+
+.background {
+  position: absolute;
+  top: 0;
+  right: 0;
+  bottom: 0;
+  left: 0;
+  width: 100vw;
+  height: calc(70vh + 56px);
```
Comment (ChristiaanScheermeijer) [inline packages/ui-react/src/components/FeaturedShelf/FeaturedShelf.module.scss]:
```suggestion
  height: calc(70vh + variables.$header-height);
```

File: packages/ui-react/src/hooks/useScrolledDown.ts (hunk 3 above)
Comment (royschut) [reply packages/ui-react/src/hooks/useScrolledDown.ts:4]: Actually, I was doing that, but that triggered lots of rerenders. So I left it.

File: packages/ui-react/src/components/FeaturedShelf/FeaturedShelf.module.scss (hunk 4 above)
Comment (royschut) [reply packages/ui-react/src/components/FeaturedShelf/FeaturedShelf.module.scss]: Yes, thanks!

File: packages/ui-react/src/hooks/useScrolledDown.ts (hunk 3 above)
Comment (ChristiaanScheermeijer) [reply packages/ui-react/src/hooks/useScrolledDown.ts:4]: I think we can do this outside React state:

**Hook interface**
```tsx
useScrollProgress(start: number, end: number, onChange: (progress: number) => void)
```

**Implementation**
```tsx
useScrollProgress(50, 600, progress => {
  posterRef.current.style.opacity = 1 - progress;
});
```

File: packages/ui-react/src/components/FeaturedShelf/FeaturedMetadata.tsx (hunk 2 above)
Comment (royschut) [reply packages/ui-react/src/components/FeaturedShelf/FeaturedMetadata.tsx:38]: I think we only want `movie` and `episode`?

File: packages/ui-react/src/hooks/useScrolledDown.ts (hunk 3 above)
Comment (royschut) [reply packages/ui-react/src/hooks/useScrolledDown.ts:4]: Yes, that should work. I will create a ticket for it.

File: packages/ui-react/src/components/FeaturedShelf/FeaturedMetadata.tsx (hunk 2 above)
Comment (royschut) [reply packages/ui-react/src/components/FeaturedShelf/FeaturedMetadata.tsx:38]: I see that `movie` is not a CONTENT_TYPE, so I just added an opt-out for `page`, `livechannel` and `hub` for now.

**Comment by royschut** [comment]
Hi @AntonLantukh and @dbudzins, I hope you find a moment to have a look as well. :-)

**Review by ChristiaanScheermeijer — APPROVED** [review]

**Pull request merged**